### ISSUE TITLE
Possibility to watch on demand?

### ISSUE THREAD

**Issue by peterdekker** [iamdevnl/plugin.video.nlziet#4]
Thanks for developing this app, watching live tv works well! I am using NLZiet to watch back/forward episodes on demand, this functionality does not seem to be supported by the app.

Would it be possible to add this in the future? Or do you have any ideas how to add this, so I could maybe look at the code myself?

Thanks!

**Comment by iamdevnl** [comment]
Hi Peter,

Good to know that the plugin is working!

Yes I am sure it is possible. I would like to add those functionality but at the moment I am a bit busy working on other projects. 
 
Not that for replay TV NLZIET is using its own service but for most on demand content NLZIET is using the external API's of NPO, RTL en Kijk.

**Comment by iamdevnl** [comment]
Hi Peter,

At the moment I am working on an integration of NPO Start, RTL XL and Kijk, so stay tuned! ;)

**Comment by peterdekker** [comment]
Great to hear! If you need any help testing, let me know! (I can also code in Python, but no prior experience with Kodi plugins)

**Comment by Jantje165** [comment]
Really good work iamdevnl! We were waiting on this addon for months.. Definitely following this :) Thank you!

**Comment by WDTY** [comment]
Working on it ;-) need to fix the DRM issues first (differ from channel implementation)

**Comment by iamdevnl** [comment]
> Working on it ;-) need to fix the DRM issues first (differ from channel implementation)

I am already implementing this functionality in this repo, should be available within a week ;)

**Comment by WDTY** [comment]
Ok that’s great news! do you need a hand? Are you going to structure the main.py into classes as wel?

**Comment by iamdevnl** [comment]
I have the functionality working to watch VOD content. I am currently working on some login optimisations, refreshing an access token etcetera. Do you have some more information about how NLZIET handles this? And yes I have restructured the whole addon with classes, I thought this was not necessary with just the live functionality.

**Comment by iamdevnl** [comment]
> Ok that’s great news! do you need a hand? Are you going to structure the main.py into classes as wel?

I have the functionality working to watch VOD content. I am currently working on some login optimisations, refreshing an access token etcetera. Do you have some more information about how NLZIET handles this? And yes I have restructured the whole addon with classes, I thought this was not necessary with just the live functionality.

**Comment by Micro-Crackedto** [comment]
Can't wait for the VOD update.
Will it auto update or do I need to redownload the addon from github manually?

**Comment by WDTY** [comment]
As for the login and oauth token routines, I just opened up chrome development tools and evaluate the calls...but then you beat me to it 😉 So I stopped analysing. What kind of problems did you ran into? For the favourites implementation I used Wouter0100 API reference: [https://github.com/Wouter0100/nlziet-api/](https://github.com/Wouter0100/nlziet-api)

**Comment by iamdevnl** [comment]
> Can't wait for the VOD update.
> Will it auto update or do I need to redownload the addon from github manually?

I am quite new in Kodi add-on development, so for now it is required to redownload the add-on from Github I think. I am investigating the possibilities for an auto update!

**Comment by iamdevnl** [comment]
> As for the login and oauth token routines, I just opened up chrome development tools and evaluate the calls...but then you beat me to it 😉 So I stopped analysing. What kind of problems did you ran into? For the favourites implementation I used Wouter0100 API reference: [https://github.com/Wouter0100/nlziet-api/](https://github.com/Wouter0100/nlziet-api)

Yeah I found that reference also but it does not include the endpoint (URL, headers etc) for refreshing the token. I am going to do some research but I will find it.

**Comment by WDTY** [comment]
Can’t check tonight, I would use the chrome developer console approach. Tomorrow evening I’m able to check also.

**Comment by WDTY** [comment]
Have you seen this repo? https://github.com/dut-iptv/dut-iptv.github.io/tree/master/matrix/plugin.video.nlziet

There could be our answers...

**Comment by eljejer** [comment]
Hi,
Unfortunately I cannot install the plugin in Kodi (Matrix) version 19.0
I get error message "The dependency on xbmc.python version 2.26.0 could not be satisfied"
Might there be a work around for this?
Apart from going to back to Kodi Leia maybe?
